android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -0,0 +1,327 @@
+        referenceEditText.setTag(R.id.repeating_group_label, repeatingGroupLabel);
+        referenceEditText.setTag(R.id.extraPopup, false);
+        referenceEditText.setTag(R.id.repeating_group_item_count, 1);
+        referenceEditText.setTag(R.id.key, "reference_edit_text");
+        referenceEditText.setTag(R.id.openmrs_entity_parent, "");
+        referenceEditText.setTag(R.id.openmrs_entity, "");
+        referenceEditText.setTag(R.id.openmrs_entity_id, "");
+
+        referenceEditText.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_FLAG_SIGNED);
+        referenceEditText.addValidator(new RegexpValidator(context.getString(R.string.repeating_group_number_format_err_msg), "\\d*"));
+        referenceEditText.addValidator(new MaxNumericValidator(context.getString(R.string.repeating_group_max_value_err_msg, MAX_NUM_REPEATING_GROUPS), MAX_NUM_REPEATING_GROUPS));
+        referenceEditText.addValidator(new MinNumericValidator(context.getString(R.string.repeating_group_min_value_err_msg), 0));
+    }
+
+    @Override
+    public List<View> getViewsFromJson(final String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener) throws Exception {
+        return getViewsFromJson(stepName, context, formFragment, jsonObject, listener, false);
+    }
+
+    private void addOnDoneAction(TextView textView, WidgetArgs widgetArgs) {
+        try {
+            InputMethodManager inputMethodManager = (InputMethodManager) widgetArgs.getFormFragment().getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
+            inputMethodManager.hideSoftInputFromWindow(textView.getWindowToken(), 0);
+            textView.clearFocus();
+            attachRepeatingGroup(textView.getParent().getParent(), Integer.parseInt(textView.getText().toString()), widgetArgs);
+        } catch (Exception e) {
+            Log.e(TAG, e.getStackTrace().toString());
+        }
+    }
+
+    private void attachRepeatingGroup(final ViewParent parent, final int numRepeatingGroups, final WidgetArgs widgetArgs) {
+
+        if (numRepeatingGroups > MAX_NUM_REPEATING_GROUPS) {
+            return;
+        }
+
+        class AttachRepeatingGroupTask extends AsyncTask<Void, Void, List<View>> {
+            private LinearLayout rootLayout = (LinearLayout) parent;
+            private List<View> repeatingGroups = new ArrayList<>();
+            private int diff = 0;
+
+            @Override
+            protected void onPreExecute() {
+                showProgressDialog(R.string.creating_repeating_group_title, R.string.creating_repeating_group_message, widgetArgs.getFormFragment().getActivity());
+            }
+
+            @Override
+            protected List<View> doInBackground(Void... objects) {
+                int currNumRepeatingGroups = ((ViewGroup) parent).getChildCount() - 1;
+                diff = numRepeatingGroups - currNumRepeatingGroups;
+                for (int i = 0; i < diff; i++) {
```

Comment:
This caters for adding new repeating groups, consider informing user that incase they reduce number of repeating groups that those will not be reflected on form